lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -309,18 +309,18 @@ def test_not_found(self, mock_request):
 @patch('requests.request')
 class SingleThreadQueryCountTestCase(ModuleStoreTestCase):
     """
-    Ensures the number of modulestore queries is deterministic based on the
-    number of responses retrieved for a given discussion thread.
+    Ensures the number of modulestore queries and number of sql queries are
+    independent of the number of responses retrieved for a given discussion thread.
     """
     MODULESTORE = TEST_DATA_MONGO_MODULESTORE
 
     @ddt.data(
-        # old mongo with cache: number of responses plus 17.  TODO: O(n)!
-        (ModuleStoreEnum.Type.mongo, 1, 23, 18),
-        (ModuleStoreEnum.Type.mongo, 50, 366, 67),
+        # old mongo with cache: 15
+        (ModuleStoreEnum.Type.mongo, 1, 21, 15, 40, 27),
+        (ModuleStoreEnum.Type.mongo, 50, 315, 15, 628, 27),
         # split mongo: 3 queries, regardless of thread response size.
-        (ModuleStoreEnum.Type.split, 1, 3, 3),
-        (ModuleStoreEnum.Type.split, 50, 3, 3),
+        (ModuleStoreEnum.Type.split, 1, 3, 3, 40, 27),
+        (ModuleStoreEnum.Type.split, 50, 3, 3, 628, 27),
```

Comment:
@muhammad-ammar - @ormsbee was doing performance investigations and discovered that the code is doing far too many SQL calls here. Do you happen to know if this change introduced the extra calls, or if this change only exposes the number of uncached calls which were already present?

@cpennington suggests that we hold an RCA to try to understand what happened here, and whether this feature introduced a performance regression. FYI @cahrens @symbolist @jimabramson.